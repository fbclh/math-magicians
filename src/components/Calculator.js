/* eslint-disable semi */
import { Component } from 'react'

class Calculator extends Component {
  constructor(props) {
    super(props)
    this.state = {}
  }

  render() {
    return (
      <div className="container">
        <div className="calculator">
          <div className="display">0</div>
          <div className="row">
            <div className="number">AC</div>
            <div className="number">+/-</div>
            <div className="number">%</div>
            <div className="symbol">รท</div>
          </div>
          <div className="row">
            <div className="number">7</div>
            <div className="number">8</div>
            <div className="number">9</div>
            <div className="symbol">x</div>
          </div>
          <div className="row">
            <div className="number">4</div>
            <div className="number">5</div>
            <div className="number">6</div>
            <div className="symbol">-</div>
          </div>
          <div className="row">
            <div className="number">1</div>
            <div className="number">2</div>
            <div className="number">3</div>
            <div className="symbol">+</div>
          </div>
          <div className="row">
            <div className="row">
              <div className="special">0</div>
            </div>
            <div className="row">
              <div className="number">.</div>
              <div className="symbol">=</div>
            </div>
          </div>
        </div>
      </div>
    )
  }
}

export default Calculator
