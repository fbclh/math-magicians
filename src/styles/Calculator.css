.container,
.calculator,
.display,
.row,
.number,
.symbol,
.special {
  display: flex;
}

.container {
  justify-content: center;
  width: 100%;
  padding: 20px;
  font-size: large;
  cursor: pointer;
}

.calculator {
  flex-direction: column;
  padding: 20px;
  text-align: center;
}

.display {
  align-items: center;
  justify-content: flex-end;
  height: 30px;
  padding: 10px;
  background-color: hsl(236, 7%, 55%);
  color: hsl(0, 0%, 100%);
}

.row {
  display: flex;
  justify-content: space-evenly;
  width: 100%;
}

.number {
  width: 70px;
}

.number,
.special {
  align-items: center;
  justify-content: center;
  height: 30px;
  padding: 10px;
  border: 1px solid hsl(0, 0%, 82%);
  background-color: hsl(0, 0%, 88%);
}

.symbol {
  align-items: center;
  justify-content: center;
  width: 70px;
  height: 30px;
  padding: 10px;
  border: 1px solid hsl(0, 0%, 82%);
  background-color: hsl(27, 90%, 60%);
}

.special {
  width: 100%;
}
