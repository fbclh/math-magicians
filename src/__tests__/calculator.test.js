import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

it('Check if Calculator page renders correctly', () => {
  const tree = renderer.create(<Calculator />);
  expect(tree).toMatchSnapshot();
});
