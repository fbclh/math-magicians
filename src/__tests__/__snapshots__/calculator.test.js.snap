// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if Calculator page renders correctly 1`] = `
<div
  className="container"
>
  <h2
    className="text"
  >
    Lets do some math!
  </h2>
  <div
    className="calculator"
  >
    <div
      className="display"
    >
       
       
    </div>
    <div
      className="row"
    >
      <div
        className="number"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex="0"
      >
        AC
      </div>
      <div
        className="number"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex="0"
      >
        +/-
      </div>
      <div
        className="number"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex="0"
      >
        %
      </div>
      <div
        className="symbol"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex="0"
      >
        รท
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="number"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex="0"
      >
        7
      </div>
      <div
        className="number"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex="0"
      >
        8
      </div>
      <div
        className="number"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex="0"
      >
        9
      </div>
      <div
        className="symbol"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex="0"
      >
        x
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="number"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex="0"
      >
        4
      </div>
      <div
        className="number"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex="0"
      >
        5
      </div>
      <div
        className="number"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex="0"
      >
        6
      </div>
      <div
        className="symbol"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex="0"
      >
        -
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="number"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex="0"
      >
        1
      </div>
      <div
        className="number"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex="0"
      >
        2
      </div>
      <div
        className="number"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex="0"
      >
        3
      </div>
      <div
        className="symbol"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex="0"
      >
        +
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="row"
      >
        <div
          className="special"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex="0"
        >
          0
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="number"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex="0"
        >
          .
        </div>
        <div
          className="symbol"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex="0"
        >
          =
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Check if Quotes page renders correctly 1`] = `
<div
  className="quotes"
>
  <p
    className="quote-p"
  >
    If I were again beginning my studies, I would follow the advice of Plato and start with mathematics. - Galileo Galilei
  </p>
</div>
`;
